samples/browser/src/main/java/org/mozilla/samples/browser/BrowserFragment.kt

```diff
@@ -114,7 +114,7 @@ class BrowserFragment : Fragment(), BackHandler {
             components.sessionManager,
             toolbar,
             sessionId
-        )
+        ) { activity?.finish() }
```

Comment:
Let's also remove the session! Otherwise it will stay somewhere in memory in the session manager :)

We'll also make sure the same happens when pressing "back" i think (although that may be out of the scope for this PR; but important).

Reply:
Oh interesting, I didn't know we needed to manually remove a session. I should also check if the sessionId needs to be cleared as well. Thanks!

Reply:
> We'll also make sure the same happens when pressing "back" i think (although that may be out of the scope for this PR; but important).

I want to handle this case (and I have an updated patch ready for it) but I don't if the CustomTabs**Toolbar**Feature is the right place. Will there be other CT features that would need to have back press handling?

Reply:
@pocmo I've updated the PR to handle back presses, but if you think it should be in another feature, I've kept in a separate commit so it can be easily removed.